orderer/common/cluster/clusterservice.go

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright IBM Corp. 2017 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"bytes"
+	"encoding/asn1"
+	"fmt"
+	"io"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+//go:generate mockery --dir . --name ClusterStepStream --case underscore --output ./mocks/
+
+// ClusterStepStream defines the gRPC stream for sending
+// transactions, and receiving corresponding responses
+type ClusterStepStream interface {
+	Send(response *orderer.ClusterNodeServiceStepResponse) error
+	Recv() (*orderer.ClusterNodeServiceStepRequest, error)
+	grpc.ServerStream
+}
+
+type AuthRequestVerifier interface {
+	VerifyAuthRequest(orderer.ClusterNodeService_StepServer, *orderer.ClusterNodeServiceStepRequest, TLSSessionBindingGetter, SignatureVerifier) (string, error)
```

Comment:
It's fine that you want to abstract out this function, but its arguments are hard to test. 

Why not input the session binding as a parameter instead of the stream? Then you have 1 parameter and not 2. 

Also, obtaining the session binding is orthogonal to verifying the request. It shouldn't be part of it.